src/routes/login/components/Login.js

```diff
@@ -0,0 +1,138 @@
+const React = require('react');
+const T = require('prop-types');
+const NavLink = require('react-router-dom').NavLink;
+const StrangeForms = require('strange-forms');
+const IsEmail = require('utils/is-email');
+
+module.exports = class Login extends StrangeForms(React.Component) {
+
+    static propTypes = {
+        errorMessage: T.string,
+        login: T.func.isRequired,
+        rememberAct: T.func.isRequired
+    };
+
+    constructor(props, context) {
+
+        super(props, context);
+
+        this.state = {
+            email: '',
+            password: '',
+            rememberMe: true,
+            isBlurred: {
+                email: false
+            }
+        };
+
+        this.strangeForm({
+            fields: ['email', 'password', 'rememberMe'],
+            get: (someProps, field) => this.state[field],
+            act: this.setFormField.bind(this),
+            getFormValue: {
+                rememberMe: this.getCheckedValue.bind(this),
+                '*': this.getFormValue.bind(this)
+            }
+        });
+    }
+
+    getCheckedValue(e) {
+
+        return e.target.checked;
+    }
+
+    getFormValue(e) {
+
+        return e.target.value || '';
+    }
+
+    setFormField(field, value) {
+
+        this.setState({ [field]: value });
+    }
+
+    fieldBlurred = (ev) => {
+
+        const isBlurred = { ...this.state.isBlurred };
+        const field = ev.target.id;
```

Comment:
Seems like this could be a potentially fragile area.  Someone changing the id of the element (which is kinda broad) might not realize it'll break the blur functionality.

Reply:
I can add some comments explaining that the id is tied to the functionality if you think that'd help.

Reply:
I would suggest leveraging javascript to capture which field was blurred, passing it directly as an argument.
```js
onBlur={() => this.fieldBlurred('email')}
```